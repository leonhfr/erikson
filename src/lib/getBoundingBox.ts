// Packages.
import bbox from '@turf/bbox';

// Internal.
import { ceiling, flooring } from './rounding';
import * as Types from '../types';

// Code.
export const getBoundingBox = (
  areaPolygon: Types.Polygon
): Types.BoundingBox => {
  const turfBoundingBox = bbox(areaPolygon);

  const [minX, minY] = turfBoundingBox.slice(0, 2).map(flooring);

  const [maxX, maxY] = turfBoundingBox.slice(2, 4).map(ceiling);

  return [minX, minY, maxX, maxY];
};
