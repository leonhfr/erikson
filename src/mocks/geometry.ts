// Internal.
import * as Types from '../types';

// Mocks.
export const polygon: Types.Polygon = {
  type: 'Feature',
  properties: {},
  geometry: {
    type: 'Polygon',
    coordinates: [
      [
        [2.1869659423828125, 41.40269443107466],
        [2.219409942626953, 41.411449504915495],
        [2.2019004821777344, 41.424322470327],
        [2.186450958251953, 41.42406503602331],
        [2.1869659423828125, 41.40269443107466],
      ],
    ],
  },
};

export const boxesPolygon: Types.Polygon = {
  type: 'Feature',
  properties: {},
  geometry: {
    type: 'Polygon',
    coordinates: [
      [
        [2, 41.58],
        [2, 41.54],
        [2.04, 41.54],
        [2.04, 41.51],
        [2.09, 41.51],
        [2.09, 41.47],
        [2, 41.47],
        [2, 41.43],
        [2.09, 41.43],
        [2.09, 41.4],
        [2.18, 41.4],
        [2.18, 41.36],
        [2.29, 41.36],
        [2.29, 41.4],
        [2.39, 41.4],
        [2.39, 41.47],
        [2.29, 41.47],
        [2.29, 41.51],
        [2.34, 41.51],
        [2.34, 41.61],
        [2.29, 41.61],
        [2.29, 41.58],
        [2.18, 41.58],
        [2.09, 41.58],
        [2, 41.58],
      ],
    ],
  },
};
