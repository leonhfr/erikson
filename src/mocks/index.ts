export * from './geometry';
export * from './sant-marti';
