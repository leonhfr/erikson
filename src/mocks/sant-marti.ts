// Packages.
import * as turfHelpers from '@turf/helpers';

// Definition
export const santMarti: turfHelpers.Feature<turfHelpers.Polygon> = {
  type: 'Feature',
  properties: {
    C_Distri: '10',
    N_Distri: 'Sant Mart√≠',
    Homes: 113572,
    Dones: 122147,
    Area: 10523761.8124,
    Perim: 20736.527911,
  },
  geometry: {
    type: 'Polygon',
    coordinates: [
      [
        [2.207193739305272, 41.427560509679019],
        [2.207150502431959, 41.427474586923964],
        [2.207117472743719, 41.427327753848751],
        [2.207057223027163, 41.427124256448593],
        [2.20702608043898, 41.427054549315891],
        [2.207005571307004, 41.42702471042724],
        [2.206852221487712, 41.426850017106617],
        [2.206842182111492, 41.426827249115867],
        [2.2068120227982, 41.426783985631516],
        [2.206453324569594, 41.426391463399192],
        [2.206334532656169, 41.426062006917604],
        [2.206286278191234, 41.425972581355602],
        [2.206151922471935, 41.425835489107683],
        [2.206050872255556, 41.425566720892334],
        [2.206153543900096, 41.425505601501307],
        [2.206238817665044, 41.425500280490368],
        [2.206361536328975, 41.425466564740233],
        [2.206458854818634, 41.42544112281842],
        [2.206881689010539, 41.425311464139966],
        [2.207253376180967, 41.42513521638918],
        [2.207446699300778, 41.425008136666143],
        [2.207771475074979, 41.424986240706019],
        [2.207847896652794, 41.424999268875318],
        [2.208099502288742, 41.425051378417756],
        [2.20843468569129, 41.425175770536683],
        [2.208491163009509, 41.425148920178458],
        [2.208746292653473, 41.425083261756377],
        [2.208997826401668, 41.425020001074977],
        [2.209121230403711, 41.424925621721854],
        [2.209150751675743, 41.424946550697349],
        [2.20916394962448, 41.424938831863642],
        [2.209210744527028, 41.424785647827953],
        [2.209349597623364, 41.424460242443274],
        [2.209411973876776, 41.424249733621167],
        [2.209594943429583, 41.424112364683921],
        [2.209620654850146, 41.424127700766306],
        [2.209700509684121, 41.424068682434978],
        [2.209804656746221, 41.424014703808012],
        [2.209903000880254, 41.423982348253929],
        [2.210254377850613, 41.423824713945756],
        [2.210394464474629, 41.42375468674912],
        [2.210537956088, 41.423661371369057],
        [2.210715728293758, 41.423528910098618],
        [2.210893365107091, 41.423350203217616],
        [2.210967752466105, 41.423266150766132],
        [2.211027360927428, 41.423174124461127],
        [2.211094237436989, 41.42295806086512],
        [2.211131766352745, 41.422802686731913],
        [2.211284164464265, 41.422336701828861],
        [2.211346307510548, 41.422171515917867],
        [2.21146795942238, 41.421848115746322],
        [2.211477113609394, 41.421819444566083],
        [2.211617845768995, 41.421378663248205],
        [2.212339842891963, 41.420833202131114],
        [2.212630590197308, 41.420799133700527],
        [2.212803036142317, 41.420747078541055],
        [2.212893171674042, 41.420720095793783],
        [2.21320711341466, 41.420543821152187],
        [2.213746518694578, 41.420278992548894],
        [2.213912283489972, 41.420187338224906],
        [2.21394506818733, 41.420154873873095],
        [2.213998856766457, 41.420132127847943],
        [2.214385555152027, 41.419891762910559],
        [2.214542901212662, 41.419780594056725],
        [2.215033465177239, 41.419409876444448],
        [2.215115059729676, 41.419345605772627],
        [2.215258387497792, 41.419232701005711],
        [2.216097040621048, 41.418600341084364],
        [2.216332107910285, 41.418450786485899],
        [2.216433223267585, 41.418375028056531],
        [2.216532326062939, 41.418299435992907],
        [2.216623801798969, 41.418221215811279],
        [2.216741788261638, 41.418102642187584],
        [2.21685121440986, 41.417978425574397],
        [2.217152819518106, 41.417587686856479],
        [2.217420152077362, 41.417283591646012],
        [2.217740791537501, 41.416951475343119],
        [2.217988585554724, 41.416627807860806],
        [2.218169958864103, 41.416490522275666],
        [2.218236230846969, 41.416447177801814],
        [2.218318357642691, 41.416360876097094],
        [2.218375622026427, 41.416313290915909],
        [2.218554265037524, 41.416185029704124],
        [2.219078888260536, 41.415872723069612],
        [2.21942398290286, 41.415600559563863],
        [2.219473522973908, 41.4155413828638],
        [2.219766438427138, 41.415245652309956],
        [2.219898690216737, 41.415075910382299],
        [2.220144456834614, 41.414689099879055],
        [2.220320882888701, 41.414454298222509],
        [2.220535632999722, 41.414241544846568],
        [2.220818114023608, 41.414011099290924],
        [2.221109153164929, 41.413871470458247],
        [2.221289150691565, 41.413741989074353],
        [2.221372430534911, 41.413633633577632],
        [2.221387192596556, 41.413570725951459],
        [2.221395523622495, 41.413526339290662],
        [2.221054529720619, 41.413152861121667],
        [2.221023485498815, 41.413091237991523],
        [2.221035301766952, 41.413071645551035],
        [2.221097007345745, 41.413027484853025],
        [2.2223029825368, 41.412519839679213],
        [2.222492305496428, 41.41244168103259],
        [2.222579517809458, 41.412405672398421],
        [2.222783978641335, 41.412321265182072],
        [2.222794548039629, 41.412316895823409],
        [2.222880297403983, 41.412160554899955],
        [2.223034359692058, 41.412091254957112],
        [2.223125760196564, 41.412050121874316],
        [2.223250691393075, 41.411987488814596],
        [2.225841304702832, 41.410935023633449],
        [2.227016600128489, 41.410457522775388],
        [2.227188067092295, 41.4103878566671],
        [2.229202926916216, 41.409569202908543],
        [2.226129212817156, 41.405281397251812],
        [2.224687441203723, 41.406348727078324],
        [2.224399004949545, 41.405912334491944],
        [2.223635949960701, 41.406198432312465],
        [2.221663347377976, 41.404627114049376],
        [2.222260208473563, 41.404185085756026],
        [2.222710182869446, 41.404921740770675],
        [2.223276159624579, 41.405426971848605],
        [2.22358899236212, 41.405180438186498],
        [2.223066554109561, 41.404698093224361],
        [2.222458453691248, 41.403696316740877],
        [2.222094896402507, 41.403401882819509],
        [2.221885216014148, 41.403571167552279],
        [2.222014660772322, 41.403803111889992],
        [2.220036151927639, 41.405271625712764],
        [2.217849526848445, 41.404050886828308],
        [2.217090286638101, 41.403039804472911],
        [2.218471129770532, 41.402117528139186],
        [2.218252287249644, 41.401932577635414],
        [2.216828981585946, 41.402883215488103],
        [2.21589083183821, 41.402168345275967],
        [2.214253041460062, 41.399987091039733],
        [2.215701478666074, 41.398914928094122],
        [2.215300615371637, 41.398618804304512],
        [2.214190550489056, 41.399440187572196],
        [2.2116367435902, 41.397493898319667],
        [2.210704240230909, 41.396251741810957],
        [2.212020654483947, 41.39521696544054],
        [2.211468936768372, 41.3948172992352],
        [2.211270182196389, 41.394673303962264],
        [2.209719550046243, 41.395886921409861],
        [2.206803062566864, 41.393773872582372],
        [2.205298363218664, 41.391631858532151],
        [2.205757891862853, 41.391220934759332],
        [2.206347091575884, 41.390774116108737],
        [2.206304849475886, 41.390742226564576],
        [2.205753075436516, 41.390325616260377],
        [2.20499801107981, 41.390904012893053],
        [2.202113197324603, 41.390003124975827],
        [2.20141228912972, 41.388783583159594],
        [2.202755678541745, 41.387771545300048],
        [2.202984601931913, 41.387943327222551],
        [2.20368494081092, 41.387417006927542],
        [2.203358562313537, 41.38687264020453],
        [2.203039060746549, 41.38641989971125],
        [2.202606915481499, 41.385887310248521],
        [2.202205356438108, 41.385446033240974],
        [2.201711536746663, 41.384969985744526],
        [2.201203630176428, 41.384537389834463],
        [2.200789373456382, 41.384218459197655],
        [2.200116319585302, 41.383805798871741],
        [2.199568232545059, 41.383530593558362],
        [2.19914504512521, 41.383848607977654],
        [2.199890004851695, 41.384199090128789],
        [2.200025654943678, 41.384152211499483],
        [2.200396976203524, 41.384381807271303],
        [2.200707168122488, 41.384606852126566],
        [2.201203038719318, 41.385020973343408],
        [2.201892700634438, 41.385675414544131],
        [2.202675829392265, 41.386591888369892],
        [2.202395022489706, 41.386803467415895],
        [2.201284820042652, 41.38596794431043],
        [2.201490176690063, 41.38581252922156],
        [2.200834828670614, 41.38531860384763],
        [2.200110702181267, 41.385864333145506],
        [2.201877365032575, 41.387193762332828],
        [2.200456383779532, 41.388265493709035],
        [2.200337634452554, 41.388176863763846],
        [2.200219539265758, 41.388265353319547],
        [2.19839545136767, 41.386893695259261],
        [2.198513277585816, 41.386803728447788],
        [2.198395393751803, 41.386715174570412],
        [2.199525133303939, 41.3858628224936],
        [2.20006279395181, 41.385178373826889],
        [2.199722962413001, 41.384934944079056],
        [2.196400865996579, 41.387414553113253],
        [2.19590447672497, 41.387534203075063],
        [2.195774073240758, 41.387886576808157],
        [2.193630215718172, 41.386333179695683],
        [2.193556521593303, 41.386289130246311],
        [2.19336228209486, 41.386421108340372],
        [2.192965382371302, 41.386729447677183],
        [2.192556711542963, 41.387034739609526],
        [2.189183853611451, 41.389618576188205],
        [2.187942429966478, 41.390569517325424],
        [2.18686574382265, 41.391408482728131],
        [2.18460019085951, 41.389755039398544],
        [2.18346004924504, 41.390610926180329],
        [2.182389633367708, 41.391425934019807],
        [2.183526574068005, 41.392258603264594],
        [2.185817357904964, 41.393936610881688],
        [2.186713515123375, 41.394593016115529],
        [2.186732443706736, 41.394904755840358],
        [2.186767748879785, 41.395486341268303],
        [2.186733831673234, 41.396631506338956],
        [2.186753821788214, 41.39668903438902],
        [2.186824900386303, 41.396893709662848],
        [2.186822363199957, 41.398266301552411],
        [2.186815898176075, 41.401764520115471],
        [2.186818893683909, 41.402287544074085],
        [2.186544627771158, 41.402276626417319],
        [2.185944774722896, 41.402439863842091],
        [2.185962974339288, 41.402465492749592],
        [2.185748893536352, 41.402543911728969],
        [2.185694280390001, 41.402640815589656],
        [2.185641908110575, 41.40272973660646],
        [2.185593211278293, 41.402816512784334],
        [2.185482225914796, 41.403037114207407],
        [2.185454397294165, 41.403105320339435],
        [2.185423871599296, 41.403198872514437],
        [2.18552228749004, 41.403317404834809],
        [2.18561624553687, 41.403659030398984],
        [2.185725669197894, 41.403830289242372],
        [2.185960142119741, 41.404071168325665],
        [2.184493331274751, 41.405179070681804],
        [2.183361528532187, 41.406032157337094],
        [2.183086414766025, 41.406243584500913],
        [2.182239849447661, 41.40689416794487],
        [2.181109959887089, 41.407749272347054],
        [2.180012604349364, 41.40857322405266],
        [2.178921951028788, 41.409432837539207],
        [2.178835942659524, 41.409469515215548],
        [2.177719363553171, 41.410300909907541],
        [2.177141540170465, 41.410747642954092],
        [2.176584163922714, 41.411179593861569],
        [2.176012404300543, 41.411629075753169],
        [2.175419055896345, 41.412039406738522],
        [2.176641905671303, 41.412949194365424],
        [2.177662396328864, 41.413706927735326],
        [2.178838461667453, 41.414558844957838],
        [2.179008517915382, 41.414681361405968],
        [2.179546737916385, 41.415075555089714],
        [2.180228978283188, 41.415575889031949],
        [2.180447948138714, 41.415860958352361],
        [2.180707331209086, 41.416031460395779],
        [2.181074913024743, 41.416306650774416],
        [2.182186701970953, 41.417115786356845],
        [2.183349053364843, 41.417956318553856],
        [2.184417987664804, 41.417152483101383],
        [2.184782091431774, 41.416869821607527],
        [2.185109543373569, 41.416624289617729],
        [2.18550409915823, 41.416336445829451],
        [2.185657240024554, 41.416209757850588],
        [2.186866098959345, 41.41528051380979],
        [2.186884067754626, 41.41526736366928],
        [2.18753295759829, 41.41479236758498],
        [2.187875867818688, 41.414516410314825],
        [2.188966218159947, 41.413666114300653],
        [2.190123041252029, 41.414554144153669],
        [2.190938552884024, 41.415189113254051],
        [2.191187455170464, 41.415382766399318],
        [2.192069080055715, 41.414770690965959],
        [2.192361632101159, 41.41456626469455],
        [2.192399475869332, 41.414655470064275],
        [2.193188650543735, 41.416904542233105],
        [2.193317260522939, 41.417169749969482],
        [2.194036192247518, 41.418972344967315],
        [2.194547605008538, 41.420535199648633],
        [2.194983458403311, 41.421733450176497],
        [2.19518796215484, 41.422361827279637],
        [2.195448850907749, 41.422332016279917],
        [2.195532165902533, 41.422454208368663],
        [2.195604467975503, 41.422650445987273],
        [2.19560506391954, 41.422696253171651],
        [2.195894687716229, 41.424449716781801],
        [2.196538412729188, 41.42658996346988],
        [2.196345964631456, 41.42670996902384],
        [2.196423627376357, 41.427071008068182],
        [2.196570098121907, 41.427401631574824],
        [2.197002318446946, 41.428238612335527],
        [2.197126146231624, 41.428476291227575],
        [2.197322839578456, 41.428853787479653],
        [2.197588412076063, 41.429170026240911],
        [2.197777714648815, 41.429356746061551],
        [2.197994640871642, 41.429523048900691],
        [2.198542243920899, 41.429817668056643],
        [2.199260839143334, 41.430057159230415],
        [2.199656804298798, 41.43013201710329],
        [2.200141461985817, 41.430166406735744],
        [2.20082915784793, 41.430128081344492],
        [2.201211785226767, 41.430058910973109],
        [2.201876992363637, 41.429879627566422],
        [2.2022944666169, 41.429721611275653],
        [2.202770380209326, 41.429566520206357],
        [2.203464341802057, 41.429277157245728],
        [2.203520937717045, 41.429253796068863],
        [2.205201011861609, 41.428479287849122],
        [2.205577163259145, 41.428305877795069],
        [2.207193739305272, 41.427560509679019],
      ],
    ],
  },
};
