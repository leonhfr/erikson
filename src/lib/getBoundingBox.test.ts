// Internal.
import { getBoundingBox } from './getBoundingBox';
import * as Mocks from '../mocks';

// Code.
describe('getBoundingBox', () => {
  it('should return the rounded bounding box', () => {
    const result = getBoundingBox(Mocks.polygon);
    expect(result).toMatchSnapshot();
  });
  it('should return the bounding box for a big polygon', () => {
    const result = getBoundingBox(Mocks.santMartiPolygon);
    expect(result).toMatchSnapshot();
  });
});
