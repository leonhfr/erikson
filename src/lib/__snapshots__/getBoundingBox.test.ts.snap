// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBoundingBox should return the rounded bounding box 1`] = `
Array [
  2.18,
  41.4,
  2.22,
  41.43,
]
`;
